kafka-ui-react-app/src/components/common/NewTable/SelectRowHeader.tsx

```diff
@@ -6,7 +6,6 @@ const SelectRowHeader: React.FC<HeaderContext<unknown, unknown>> = ({
   table,
 }) => (
   <IndeterminateCheckbox
-    checked={table.getIsAllPageRowsSelected()}
```

Comment:
I don't think this is the correct solution you just removed the props from the common component , this change will break the check when all the items are selected the header will not show the check.

in `SelectRowHeader`  debug onChange function

```
const SelectRowHeader: React.FC<HeaderContext<unknown, unknown>> = ({
  table,
}) => {
  const c = (a: unknown) => {
    console.log('something', table.getIsAllPageRowsSelected());
    table.getToggleAllPageRowsSelectedHandler()(a);
  };

  return (
    <IndeterminateCheckbox
      checked={table.getIsAllPageRowsSelected()}
      indeterminate={table.getIsSomePageRowsSelected()}
      onChange={c}
    />
  );
};
```

check the result in the working case , vs non working case where the result in console stays false. we need to check Tanstack table documentation to see how to navigate around this problem

Reply:
I have thought  we don't need default value on header checkbox

Reply:
`table.getIsAllRowsSelected()` yep this is the correct config